Mathlib/Algebra/Homology/ShortComplex/ConcreteCategory.lean

```diff
@@ -0,0 +1,77 @@
+/-
+Copyright (c) 2023 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.Algebra.Homology.ShortComplex.Ab
+
+/-!
+# Exactness of short complexes in concrete abelian categories
+
+If an additive concrete category `C` has an additive forgetful functor to `Ab`
+which preserves homology, then a short complex `S` in `C` is exact
+if and only if it is so after applying the functor `forget₂ C Ab`.
+
+-/
+
+namespace CategoryTheory
+
+variable {C : Type*} [Category C] [ConcreteCategory C] [HasForget₂ C Ab]
+
+@[simp]
+lemma ShortComplex.zero_apply
+    [Limits.HasZeroMorphisms C] [(forget₂ C Ab).PreservesZeroMorphisms]
+    (S : ShortComplex C) (x : (forget₂ C Ab).obj S.X₁) :
+    ((forget₂ C Ab).map S.g) (((forget₂ C Ab).map S.f) x) = 0 := by
+  erw [← comp_apply, ← Functor.map_comp, S.zero, Functor.map_zero]
+  rfl
+
+section abelian
+
+variable [Abelian C] [(forget₂ C Ab).Additive] [(forget₂ C Ab).PreservesHomology]
```

Comment:
Do you need the full force of abelian categories here?

Reply:
Thanks for the suggestion! Indeed, `[Preadditive C]` and `[HasZeroObject C]` are sufficient here.